luarules/gadgets/game_disable_assist_ally.lua

```diff
@@ -17,10 +17,10 @@ if not gadgetHandler:IsSyncedCode() then
 	return false
 end
 
-local disable_assist = Spring.GetModOptions().disable_assist_ally_construction
+local disableAssist = Spring.GetModOptions().disable_assist_ally_construction
 
-if not disable_assist then
-	return false
+if disableAssist then
+	return true
 end
```

Comment:
I don't think this this does what you intend. If the modoption to disable assisting is set, it will return early and not run any of the other code in the file, which will not disable assisting.
```suggestion
if allowAssist then
	return false
end
```

Reply:
Ah you're right I should've tested that thanks.